Sources/Basics/HTTPClient.swift

```diff
@@ -0,0 +1,262 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+#if canImport(FoundationNetworking)
+// TODO: this bring OpenSSL depeendcy on Linux
+// need to decide how to best deal with that
+import FoundationNetworking
+#endif
+
+// MARK: - HTTPClient
+
+public protocol HTTPClient {
```

Comment:
What do you think about putting this in TSCBasic since this is completely generic API?

Reply:
That's a good idea. I suggest we continue to bake it in SwiftPM until we have all the key features in and then extract out to TSC